src/Factory.php

```diff
@@ -440,4 +479,56 @@ private function isPersisting(): bool
             return false;
         }
     }
+
+    private function resetDoctrineEvents(): void
+    {
+        $this->doctrineEvents = true;
+        $this->doctrineEventTypeName = null;
+
+        /** @var EntityManagerInterface $manager */
+        $manager = self::configuration()->objectManagerFor($this->class);
+        /** @var ContainerAwareEventManager $eventManager */
+        $eventManager = $manager->getConnection()->getEventManager();
+
+        foreach ($this->disabledDoctrineEvents as $type => $listenersByType) {
+            foreach ($listenersByType as $listener) {
+                if ($listener instanceof EventSubscriber) {
+                    $eventManager->addEventSubscriber($listener);
+
+                    continue;
+                }
+
+                $eventManager->addEventListener($type, $listener);
+            }
+        }
+    }
+
+    private function disableDoctrineEvents(): void
+    {
+        /** @var EntityManagerInterface $manager */
+        $manager = self::configuration()->objectManagerFor($this->class);
+        /** @var ContainerAwareEventManager $eventManager */
+        $eventManager = $manager->getConnection()->getEventManager();
+
+        $this->disabledDoctrineEvents = $manager->getConnection()->getEventManager()->getListeners($this->doctrineEventTypeName);
+        foreach ($this->disabledDoctrineEvents as $type => $listenersByType) {
+            foreach ($listenersByType as $name => $listener) {
+                if ($listener instanceof EventSubscriber) {
+                    $manager->getConnection()->getEventManager()->removeEventSubscriber($listener);
+
+                    continue;
+                }
+
+                if (false === \mb_strpos($name, '_service_')) {
+                    // Handle listeners with hash as name, ex: '000000001f7b9da2000000001c719ccc'
+                    $eventManager->removeEventListener($type, $listener);
```

Comment:
This is well tested because in the test I call twice `withoutDoctrineEvents()`, when the `create()` method of the first one finish, it re enable listeners and register them with that kind of name.  
So the second  `withoutDoctrineEvents()` test this part.